tests/test_database.py

```
@@ -0,0 +1,572 @@
+        mock_connection = AsyncMock()
+        mock_connection.fetchrow = AsyncMock(return_value=None)
+
+        mock_pool = AsyncMock(spec=asyncpg.Pool)
+        mock_pool.acquire.return_value.__aenter__.return_value = mock_connection
+        db_manager.pool = mock_pool
+
+        result = await db_manager.fetchrow("SELECT * FROM test WHERE id = $1", 999)
+
+        assert result is None
+        # Verify debug log for no rows
+        debug_calls = [call for call in mock_logger.debug.call_args_list]
```

Ruff (0.14.5)

Unnecessary list comprehension (rewrite using `list()`)

Rewrite using `list()`

(C416)